From 16d5df20c71c97560e44b45e23bc7946cbb8668f Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Thu, 24 Dec 2020 18:47:56 +0800
Subject: [PATCH] media-libva: Add export AYUV color in vaExportSurfaceHandle
 when VA_EXPORT_SURFACE_COMPOSED_LAYERS is not set.

Example:
gst-launch-1.0 videotestsrc ! msdkvpp \
'video/x-raw(memory:DMABuf)',format=VUYA ! glimagesink

fixes #1114

Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>

Upstream-status: backport
https://github.com/intel/media-driver/commit/ff0cb9efc9f30712e8b967da629b2804c9a128a1

Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 media_driver/linux/common/ddi/media_libva.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media_driver/linux/common/ddi/media_libva.cpp b/media_driver/linux/common/ddi/media_libva.cpp
index 875603ad..93add4cd 100755
--- a/media_driver/linux/common/ddi/media_libva.cpp
+++ b/media_driver/linux/common/ddi/media_libva.cpp
@@ -6679,6 +6679,8 @@ static uint32_t DdiMedia_GetDrmFormatOfSeparatePlane(uint32_t fourcc, int plane)
             return DRM_FORMAT_VYUY;
         case VA_FOURCC_UYVY:
             return DRM_FORMAT_UYVY;
+        case VA_FOURCC_AYUV:
+            return DRM_FORMAT_AYUV;
         case VA_FOURCC_Y210:
             return DRM_FORMAT_Y210;
         case VA_FOURCC_Y216:
-- 
2.17.1

